recipes/caf/all/conanfile.py

```diff
@@ -0,0 +1,132 @@
+import os
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
+
+
+class CAFConan(ConanFile):
+    name = "caf"
+    description = "An open source implementation of the Actor Model in C++"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/actor-framework/actor-framework"
+    topics = "conan", "caf", "actor-framework", "actor-model", "pattern-matching", "actors"
+    license = "BSD-3-Clause", "BSL-1.0"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "log_level": ["ERROR", "WARNING", "INFO", "DEBUG", "TRACE", "QUIET"],
+        "with_openssl": [True, False]
+    }
+    default_options = {"shared": False, "fPIC": True, "log_level": "QUIET", "with_openssl": True}
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+    _cmake = None
+
+    @property
+    def _has_openssl(self):
+        return 'with_openssl' in self.options.values.keys() and self.options.with_openssl
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+            if self.settings.arch == "x86":
+                del self.options.with_openssl
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("actor-framework-" + self.version, self._source_subfolder)
+
+    def requirements(self):
+        if self._has_openssl:
+            self.requires("openssl/1.1.1j")
+
+    def configure(self):
+        if self.options.shared and self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("Shared libraries are not supported on Windows")
+        compiler_version = Version(self.settings.compiler.version.value)
+        if Version(self.version) > "0.17.6" and \
```

Comment:
Maybe copy conan-io/conan#8002 and just add the few minor exceptions?

Reply:
Does `tools.check_min_cppstd` work if `compiler.cppstd` is not set?  The docs make it appear so:

    Validates if the applied cppstd setting (from compiler.cppstd settings or 
    deducing the default from compiler and compiler.version) is at least the 
    value specified

If it does then is conan-io/conan#8002 needed?  Or does “the default” mean the default compiler language level, not what it’s capable of?

Reply:
> the default mean the default compiler language level, not what it’s capable of?

Thats correct, and to compound the issue CCI does not set the value so there's very little control and often more challenges since the apple-clang default is C++98 until version 12+

Reply:
I agree that conan-io/conan#8002 is totally needed then.  

In this case we would need the recipe to support both C++11 and C++17.  And there's one manual check that needs to be done because of a bug in Clang 5 that fails to build `test_package`.

I will add this.  I don't think we can just use `self.compiler.get_safe("cppstd", std)` because that is not used in profiles universally.